### PR TITLE
To Do list: Milestone 3 interactive list

### PR DESCRIPTION
**Pull request by MNisarAli**
## To-Do list: interactive list functionality

In this project I implemented the following as per the requirements:

- Added a new JavaScript file for task status (completed: true/false). and imported it as a module.    
- Added event listener to the checkbox (`change`).
- Update items object's value for the `completed` key upon user actions.
- Implemented a function for the "Clear all completed" button using the `filter()` method.
- Stored the updated array of items in local storage, so the user can get the correct list values after the page reloads.
- Updated HTML, and CSS files.
- All source files generated by Webpack.

## General requirements checklist.

- [X] No linters error.
- [X] Followed the GitHub flow.
- [X] documented work professionally.
- [X] Followed the best practices of HTML, CSS, and JavaScript.

### CODE REVIEW

File: src/userActions.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+export default class UserActions {
+  // Mark Task As Completed & Update Its Object's Value For Completed Key.
+  completeTask = (items) => {
```
Comment (muneebulrehman) [inline src/userActions.js:3]: Upon reloading the page, the tasks which are marked complete don't show such for their checkboxes. Kindly fix this issue. I recorded [this video](https://www.loom.com/share/585bffd6db9d4f3e8c575f7326d3ea88) where I demonstrated the issue.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @MNisarAli , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/taskUpdate.js [hunk 2]
```diff
@@ -1,10 +1,23 @@
+import UserActions from './userActions.js';
+
+const userAction = new UserActions();
+
 export default class Tasks {
```
Comment (muneebulrehman) [inline src/taskUpdate.js:5]: After adding some tasks, marking some complete, then clearing the completed tasks, followed by adding some tasks again, all without reloading, there seems to be a duplication of indexes as you can see in the screenshot of the local storage. Kindly fix this issue.
image

Comment (MNisarAli) [reply src/taskUpdate.js:5]: I have made changes. Thanks for your review.

File: src/userActions.js (hunk 1 above)
Comment (MNisarAli) [reply src/userActions.js:3]: I have made the changes. Thanks for your review.

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @MNisarAli ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**